src/components/PopupConfirm.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector, handleButtonClick) {
+    super(popupSelector);
+    this._handleButtonClick = handleButtonClick;
+    this._button = this._popupEl.querySelector(".modal__button");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._button.addEventListener("click", () => {
+      this._handleButtonClick(this._calledEl);
+    });
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`